triss-core/src/main/kotlin/com/pp/trisscore/service/ComparisonService.kt

```diff
@@ -12,8 +12,19 @@ import org.springframework.stereotype.Service
 @Service
 class ComparisonService {
 
-    fun compareApplicationsInfo(dbApplicationInfo: ApplicationInfo, reqApplicationInfo: ApplicationInfo, role: Role) {
-        compareApplications(dbApplicationInfo.application, reqApplicationInfo.application, role)
+    fun compareApproveApplicationsInfo(dbApplicationInfo: ApplicationInfo, reqApplicationInfo: ApplicationInfo, role: Role) {
+        compareApplications(dbApplicationInfo.application, reqApplicationInfo.application,role)
+        compareApproveApplication(role, dbApplicationInfo.application, reqApplicationInfo.application)
+        compareInstitutes(dbApplicationInfo.institute, reqApplicationInfo.institute)
+        if (role != Role.DIRECTOR)
+            compareFinancialSources(dbApplicationInfo.financialSource!!, reqApplicationInfo.financialSource!!)
+        compareAdvanceApplication(dbApplicationInfo.advanceApplication, reqApplicationInfo.advanceApplication)
+        compareAdvancePayments(dbApplicationInfo.advancePayments, reqApplicationInfo.advancePayments)
+    }
+
+    fun compareRejectedApplicationsInfo(dbApplicationInfo: ApplicationInfo, reqApplicationInfo: ApplicationInfo, role: Role) {
+        compareApplications(dbApplicationInfo.application, reqApplicationInfo.application,role)
+        compareRejectedApplications(role, dbApplicationInfo.application, reqApplicationInfo.application)
```

Comment:
I don't like that in "compareApplications" role argument is in 3 place, and in "compareRejectedApplications" and "compareApprovedApplication" is in first place